lib/cli_util.dart

```diff
@@ -57,4 +57,18 @@ Directory getSdkDir([List<String> cliArgs]) {
 }
 
 /// Return the path to the current Dart SDK.
-String getSdkPath() => path.dirname(path.dirname(Platform.resolvedExecutable));
+String getSdkPath() {
+  // Handle the case of a finished dart-sdk
+  var fromSdkPath = path.dirname(path.dirname(Platform.resolvedExecutable));
+  if (isSdkDir(Directory(fromSdkPath))) {
+    return fromSdkPath;
+  }
+  // Handle the case where Platform.executable is a sibling of the SDK directory
```

Comment:
I'm not sure whether we should support this case. We'll want to assume that we're running from a well-formed dart sdk; this is an edge case specifically for some cases running from the dart-lang/sdk repo.

What's the use case we're trying to make work here? Can we switch the internal testing use to use the fully formed sdk in out/ReleaseX64/dart-sdk?

Alternatively, we could in-line this specific logic into the one or two place where this is needed (in-line `getSdkPath()`) instead of putting this in a general purpose package.

Reply:
If we really need some operation to be performed without the correct SDK structure, then this operation should have this work-around. But the general purpose package should support only the correct SDK structure. And even better, even in case when we cannot have full SDK, just put what you have into right locations. Last time I was told that some AOT snapshots might be not ready, well, then put them later, but keep everything else in place.